posthog/models/test/test_async_deletion_model.py

```diff
@@ -0,0 +1,291 @@
+from uuid import UUID, uuid4
+
+from posthog.client import sync_execute
+from posthog.models import AsyncDeletion, DeletionType, Team, User
+from posthog.models.async_deletion.delete import mark_deletions_done, run_event_table_deletions
+from posthog.models.cohort.util import insert_static_cohort
+from posthog.models.group.util import create_group
+from posthog.models.person.util import create_person, create_person_distinct_id
+from posthog.models.plugin import PluginLogEntrySource, PluginLogEntryType
+from posthog.test.base import (
+    BaseTest,
+    ClickhouseDestroyTablesMixin,
+    ClickhouseTestMixin,
+    _create_event,
+    snapshot_clickhouse_alter_queries,
+    snapshot_clickhouse_queries,
+)
+from posthog.test.test_plugin_log_entry import create_plugin_log_entry
+
+uuid = str(UUID("6ba7b810-9dad-11d1-80b4-00c04fd430c8"))
+uuid2 = str(UUID("7ba7b810-9dad-11d1-80b4-00c04fd430c8"))
+
+
+class TestAsyncDeletion(ClickhouseTestMixin, ClickhouseDestroyTablesMixin, BaseTest):
+    def setUp(self):
+        super().setUp()
+        self.user = User.objects.create(email="test@posthog.com")
+        self.teams = [
+            self.team,
+            Team.objects.create(organization=self.organization),
+            Team.objects.create(organization=self.organization),
+        ]
+
+    @snapshot_clickhouse_queries
+    def test_mark_team_deletions_done(self):
+        deletion = AsyncDeletion.objects.create(
+            deletion_type=DeletionType.Team, team_id=self.teams[0].pk, key=str(self.teams[0].pk), created_by=self.user
+        )
+
+        mark_deletions_done()
+
+        deletion.refresh_from_db()
+        self.assertIsNotNone(deletion.delete_verified_at)
+
+    @snapshot_clickhouse_queries
+    def test_mark_team_deletions_done_when_not_done(self):
```

Comment:
following the pattern you use below

```suggestion
    def test_mark_deletions_done_team_when_not_done(self):
```